examples/README.md

```diff
@@ -1,17 +0,0 @@
-# Crossplane Examples
```

Comment:
Can we link to `examples` folder of providers somewhere? Either in `README.md` or `examples/README.md`? I believe the shortest path to the actual examples should be as short as possible. An `examples` folder in the root of the repo is pretty good in that sense but if we remove it, at least let's make sure we have references to examples in READMEs and documentation website.

Reply:
There are also some docs links that I think we may need to clean up as a result of this

Reply:
What do you guys think about having only `examples/README.md` that points to `examples` folder of the providers one by one?

Reply:
This sounds good to me, but I'd like to keep this diff mostly red and handle that in a follow-on PR.

Reply:
Just cleaned up any references from the docs to the examples, by moving the referenced examples (scripts, actually) into docs/snippets.